src/main/java/com/indeed/jiraactions/api/ApiCaller.java

```diff
@@ -3,178 +3,130 @@
+                        value = String.join(",", requestHeaders.value(i));
                     }
                     sb.append("\"").append(key).append("\": \"").append(value).append("\",");
                 }
-            }
-            sb.append("}");
+                sb.append('}');
+                sb.append('}');
 
-            sb.append("}");
-
-            if (urlConnection != null) {
                 sb.append(", \"Response\": {");
-                for (final Map.Entry<String, List<String>> header : urlConnection.getHeaderFields().entrySet()) {
-                    final String key = header.getKey();
+                for (int i = 0, size = responseHeaders.size(); i < size; i++) {
+                    final String key = responseHeaders.name(i);
                     final String value;
                     if ("Set-Cookie".equals(key)) {
                         value = "<Omitted>";
                     } else {
-                        value = String.join(",", header.getValue());
+                        value = String.join(",", responseHeaders.value(i));
                     }
                     sb.append("\"").append(key).append("\": \"").append(value).append("\",");
                 }
-                sb.append("\"Code\": ").append(urlConnection.getResponseCode()).append(",");
-                sb.append("\"Message\": \"").append(urlConnection.getResponseMessage()).append("\",");
-            }
-            if (apiResults != null) {
-                sb.append("\"Body\": \"").append(apiResults).append("\"");
-            }
-            if (urlConnection != null) {
-                final InputStream error = urlConnection.getErrorStream();
-                if(error != null) {
-                    final BufferedReader errorBr = new BufferedReader(new InputStreamReader(error));
-                    sb.append("\"Error Body\": \"");
-                    String line = errorBr.readLine();
-                    while (line != null) {
-                        sb.append(line).append(System.lineSeparator());
-                        line = errorBr.readLine();
-                    }
-                    errorBr.close();
-                } else {
-                    sb.append("Unable to open error stream for reading");
-                }
-                sb.append("\"");
+                sb.append("\"Code\": ").append(response.code()).append(",");
+                sb.append("\"Message\": \"").append(response.message()).append("\",");
+                sb.append("\"Error Body\": \"").append(responseBody.string()).append("\"");
+                sb.append('}');
+                sb.append('}');
+                throw new IOException(String.valueOf(response));
             }
-
-            sb.append("}");
-            sb.append("}");
-            log.error("Encountered connection error: " + sb);
+            return objectMapper.readTree(responseBody.string());
+        } catch (IOException e) {
```

Comment:
Why are we catching this exception only to rethrow it?

Reply:
I modeled this kind of similar to the existing code so it logs the stringbuilder. Should we just not do that and log the error instead?